WordPress/gutenberg
# Quick edit: add Date as field and `datetime` as field type

Part of https://github.com/WordPress/gutenberg/issues/59745 https://github.com/WordPress/gutenberg/issues/63624

## What?

Adds the "Date" field in quick edit as a `datetime` type.

<img width="2076" alt="Captura de ecrã 2024-08-07, às 08 38 33" src="https://github.com/user-attachments/assets/5756b1f6-16ae-4f52-a9c4-74c6ae2c46bb">

https://github.com/user-attachments/assets/2aae36ab-1765-4e2e-8c63-60334bcf0b73


## How?

This PR introduces a `datetime` field type definition to be used by the "Date" field.

## Testing Instructions

- With the "quick edit" experiment enabled, visit the Pages page and switch to table.
- Open the quick edit form and select a record.
- Verify there's a new date-time component and that it updates the item.
